java-frontend/src/main/java/org/sonar/java/se/NullableAnnotationUtils.java

```diff
@@ -71,22 +83,32 @@ private static boolean isAnnotatedWithOneOf(Symbol symbol, Set<String> annotatio
   }
 
   public static boolean isGloballyAnnotatedParameterNonNull(Symbol.MethodSymbol method) {
-    return isGloballyAnnotatedWith(method, "javax.annotation.ParametersAreNonnullByDefault")
-      || usesEclipseNonNullByDefault(method, "PARAMETER");
+    return parameterNonNullGlobalAnnotation(method).isPresent();
+  }
+
+  public static Optional<String> parameterNonNullGlobalAnnotation(Symbol.MethodSymbol method) {
```

Comment:
as far as I understand the usage of this method it seems you don't get anything out of Optional: this method should just return null